checkin_collection.json

```diff
@@ -0,0 +1,104 @@
+            "type": "integer"
+          },
+          "pieces": {
+            "description": "A collection of piece records",
+            "type": "array",
+            "id": "pieces",
+            "piece": {
+              "type": "object",
+              "properties": {
+                "id": {
+                  "description": "The id of the piece",
+                  "type": "string",
+                  "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+                },
+                "barcode": {
+                  "description": "The barcode assigned to the piece",
+                  "type": "string"
+                },
+                "comment": {
+                  "description": "The free form notes pertaining to the piece",
+                  "type": "string"
+                },
+                "caption": {
+                  "description": "The enumeration caption of the piece",
+                  "type": "string"
+                },
+                "create_item": {
+                  "description": "Whether or not to create an item record for this piece",
+                  "type": "boolean"
+                },   
+                "supplement": {
+                  "description": "Whether or not this is a supplementary material for this piece",
+                  "type": "boolean"
+                }, 
+                "location_id": {
+                  "description": "The id of the location",
+                  "type": "string",
+                  "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+                },
+                "material_type_id": {
+                  "description": "The id of the materialType",
+                  "type": "string",
+                  "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+                },
+                "product_type": {
+                  "description": "The type of the Product",
+                  "type": "string"
+                },
+                "product_type_id": {
+                  "description": "The id of the productType",
+                  "type": "string",
+                  "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+                },
+                "accession_number": {
+                  "description": "The number referencing physical item acquired by the library",
+                  "type": "string"
+                },
+                "item_description": {
+                  "description": "The description associated with the item record",
+                  "type": "string"
+                },
+                "electronic_bookplate": {
+                  "description": "A text that relates to the owner of the book",
+                  "type": "string"
+                }                                
```

Comment:
Please add here:
```json
              "required": [
                "caption",
                "location_id"
              ]
```
In this case the jaxrs-code-generator will be able to add `@NotNull` annotation in the model for the corresponding properties.